### DISCUSSION TITLE
systemd: what's in your .sbclrc?

### DISCUSSION THREAD

**Discussion by vindarel** [rabbibotton/clog#143] [Q&A] [answered]
Hello,

In this tutorial: https://github.com/rabbibotton/clog/blob/main/WEBSERVER.md you tell SBCL to read your .sbclrc. Can we see what you initialize in there? I think it has its importance to require asdf, load quicklisp and more.

Context: I'm currently having an issue starting a CL web service with systemd (but not CLOG).

Thanks

**Comment by rabbibotton** [comment, marked as answer]
Just the quickslip additons

;;; The following lines added by ql:add-to-init-file:
#-quicklisp
(let ((quicklisp-init (merge-pathnames ".quicklisp/setup.lisp"
                                       (user-homedir-pathname))))
  (when (probe-file quicklisp-init)
    (load quicklisp-init)))

**Reply by vindarel** [reply]
ahah, I just found also that you tell SBCL to use it with `--userinit path/to/.sbclrc`. At first I didn't, so the snippet was looking for quicklisp in my root directory (`user-homedir-pathname` and I run systemd as root, so, boggus).

**Comment by vindarel** [comment]
I had to `(require 'asdf)`. Already happened in a CI. Not sure why.

And as says your tutorial, it is important to `sleep` after the web server is started, otherwise Lisp exits and Systemd sees it as a fatal error (there's a non-0 exit code I suppose). I used:

```
(bt:join-thread (find-if (lambda (th)
                               (search "hunchentoot" (bt:thread-name th)))
                             (bt:all-threads)))
```
as in https://lispcookbook.github.io/cl-cookbook/scripting.html#for-web-apps

Thanks for the forum!

**Reply by rabbibotton** [reply]
That works too, I prefer to do the (loop (sleep 360)) as I am a control freak :) and to remind me that I should place some code in there to monitor things in the future :)
